table/block_based_table_reader.cc

```diff
@@ -2471,12 +2502,9 @@ Status BlockBasedTable::Prefetch(const Slice* const begin,
   for (begin ? iiter->Seek(*begin) : iiter->SeekToFirst(); iiter->Valid();
        iiter->Next()) {
     Slice block_handle = iiter->value();
-    const bool is_user_key = rep_->table_properties &&
-                             rep_->table_properties->index_key_is_user_key > 0;
-    if (end &&
-        ((!is_user_key && comparator.Compare(iiter->key(), *end) >= 0) ||
-         (is_user_key &&
-          user_comparator->Compare(iiter->key(), ExtractUserKey(*end)) >= 0))) {
+    std::string ikey;
+    AppendInternalKey(&ikey, iiter->parsed_internal_key());
```

Comment:
I am concerned about the overhead of the added memcpy for each comparison. I am not sure if this change is essential to this PR.